src/EditorFeatures/Core.Wpf/InlineRename/InlineRenameSession.cs

```diff
@@ -741,11 +762,25 @@ private void ApplyRename(Solution newSolution, IWaitContext waitContext)
                         var newText = newDocument.GetTextAsync(waitContext.CancellationToken).WaitAndGetResult(waitContext.CancellationToken);
                         finalSolution = finalSolution.WithDocumentText(id, newText);
                     }
+
+                    // Make sure to include any document rename as well
+                    finalSolution = finalSolution.WithDocumentName(id, newDocument.Name);
```

Comment:
the premise here is that this is a no-op if the name didn't change?

Reply:
Correct, and I double checked that it is